<?xml version="1.0" ?>
<launch>
  
  <arg name="sim" default="false" />
  <arg unless="$(arg sim)" name="kinfu_cloud_topic_arg" value="/ros_kinfu/depth_registered/points" />
  <arg     if="$(arg sim)" name="kinfu_cloud_topic_arg" value="/camera/depth_registered/points_" />

  <arg unless="$(arg sim)" name="capsicum_image_topic_arg" value="/camera/rgb/image_raw" />
  <arg     if="$(arg sim)" name="capsicum_image_topic_arg" value="/camera/rgb/image_raw_" />

  <node pkg="fruit_picking" type="field_trial_2" name="field_trial_2" output="screen">

    <!-- INPUT Subscriber Topics -->
    <param name="scan_delta_x" value="0.065" />
    <param name="scan_delta_y" value="0.07" />
    <param name="scan_delta_z" value="0.07" />

    <param name="start_x" value="0.15" />
    <param name="start_y" value="-0.52" />
    <param name="start_z" value="1.1" />

    <param name="start_q1" value="1.0" />
    <param name="start_q2" value="0.0" />
    <param name="start_q3" value="0.0" />
    <param name="start_q4" value="0.0" />

    <param name="velocity_scan" value="0.1" />
    <param name="velocity_attach" value="0.25" />
    <param name="velocity_cut" value="0.05" />

    <param name="wait_time" value="0.0" />
    <param name="plan_attempts" value="2" />

    <param name="scan_offset" value="0.3" />

    <param name="tool_point_x" value="0.37" />
    <param name="tool_point_y" value="0.0" />
    <param name="tool_point_z" value="0.06" />
    <param name="tool_point_offset_x" value="0.2" />

    <param name="scan_offset_x" value="0.37" />
    <param name="scan_offset_y" value="0.0" />
    <param name="scan_offset_z" value="0.1" />

    <param name="gripper_pose_topic" value="gripper_pose" />
    <param name="kinfu_reset_topic" value="/ros_kinfu/reset" />
    <param name="superquadric_topic" value="/superquadric_fitter/estimate_capsicum_pose" />
    <param name="ur_set_io_topic" value="/set_io" />
    <param name="ur_io_states_topic" value="/io_states" />
    <param name="segment_capsicum_topic" value="/capsicum_detection/segment_capsicum"/>

    <param name="capsicum_cloud_topic" value="/capsicum/points" />

    <param name="kinfu_cloud_topic" value = "$(arg kinfu_cloud_topic_arg)"/>
    <param name="capsicum_image_topic" value="$(arg capsicum_image_topic_arg)"/>


  </node>


</launch>
